pkg/controller/recoveries.go

```diff
@@ -151,9 +151,31 @@ func (c *StashController) runRecoveryInjector(key string) error {
 	if !exists {
 		// Below we will warm up our cache with a Recovery, so that we will see a delete for one d
 		fmt.Printf("Recovery %s does not exist anymore\n", key)
-	} else {
-		d := obj.(*api.Recovery)
-		fmt.Printf("Sync/Add/Update for Recovery %s\n", d.GetName())
+		return nil
 	}
+
+	d := obj.(*api.Recovery)
+	fmt.Printf("Sync/Add/Update for Recovery %s\n", d.GetName())
+	return c.runRecoveryJob(d)
+}
+
+func (c *StashController) runRecoveryJob(rec *api.Recovery) error {
+	restic, err := c.stashClient.Restics(rec.Namespace).Get(rec.Spec.Restic, metav1.GetOptions{})
+	if err != nil {
+		log.Infoln(err)
+		return err
+	}
+
+	if err = restic.IsValid(); err != nil {
+		log.Infoln(err)
+		return err
+	}
+
+	job := util.CreateRecoveryJob(rec, restic, c.options.SidecarImageTag)
```

Comment:
We should check 2 things:
- Recovery Status != RUNNING or != SUCCEEDED.
- Job with the same name does not exist.

We should use Job name like `stash-{recovery-name}` to avoid potential conflict with user job.